.github/workflows/test_rocprim.yml

```diff
@@ -0,0 +1,92 @@
+# Copyright 2025 Advanced Micro Devices, Inc.
+#
+# Licensed under the Apache License v2.0 with LLVM Exceptions.
+# See https://llvm.org/LICENSE.txt for license information.
+# SPDX-License-Identifier: Apache-2.0 WITH LLVM-exception
+
+name: rocPRIM test
+
+on:
+  workflow_dispatch:
+    inputs:
+      artifact_run_id:
+        type: string
+        default: ""
+      amdgpu_target:
+        type: string
+        default: gfx94X-dcgpu
+  workflow_call:
+    inputs:
+      artifact_run_id:
+        type: string
+        default: ""
+      amdgpu_target:
+        type: string
+        default: gfx94X-dcgpu
+  push:
+    branches:
+      - ADHOCBUILD
+
+jobs:
+  test_rocprim:
+    name: "Test rocPRIM"
+    runs-on: linux-mi300-1gpu-ossci-rocm
+    strategy:
+      fail-fast: true
+    defaults:
+      run:
+        shell: bash
+    env:
+      VENV_DIR: ${{ github.workspace }}/.venv
+      ARTIFACT_RUN_ID: "${{ inputs.artifact_run_id != '' && inputs.artifact_run_id || github.run_id }}"
+      BUILD_ARTIFACTS_DIR: "build/artifacts"
+
+    # MOSTLY BOILER PLATE ABOVE.
+    steps:
+      - name: Checkout Repository
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+
+      - name: Install the AWS tool
+        run: ./dockerfiles/cpubuilder/install_awscli.sh
```

Comment:
Feels a bit weird to use a dockerfile script outside of Docker. That script might make future assumptions about the environment that won't work from a github action. Could use https://github.com/marketplace/actions/install-aws-cli-action or some other steps instead.